tests/integration/techRecords.intTest.ts

```diff
@@ -290,6 +294,237 @@ describe("techRecords", () => {
+                request.post("vehicles")
+                  .send(newTechRec)
+                  .end((err, res: any) => {
+                    expect(res.statusCode).toEqual(201);
+                    expect(res.headers["access-control-allow-origin"]).toEqual("*");
+                    expect(res.headers["access-control-allow-credentials"]).toEqual("true");
+                    expect(res.body).toEqual("Technical Record created");
+                    done();
+                  });
+              });
+            });
+
+            context("and that vehicle does exist", () => {
+              it("should return status 400 The conditional request failed", (done) => {
+                request.post("vehicles")
+                  .send(newTechRec)
+                  .end((err, res: any) => {
+                    expect(res.statusCode).toEqual(400);
+                    expect(res.headers["access-control-allow-origin"]).toEqual("*");
+                    expect(res.headers["access-control-allow-credentials"]).toEqual("true");
+                    expect(res.body).toEqual("The conditional request failed");
+                    done();
+                  });
+              });
+            });
+          });
+
+          context("and the payload is invalid", () => {
+            context("and the techRecord array is empty", () => {
+              it("should return status 400 The conditional request failed", (done) => {
+                newTechRec.techRecord = [];
+                request.post("vehicles")
+                  .send(newTechRec)
+                  .end((err, res: any) => {
+                    expect(res.statusCode).toEqual(400);
+                    expect(res.headers["access-control-allow-origin"]).toEqual("*");
+                    expect(res.headers["access-control-allow-credentials"]).toEqual("true");
+                    expect(res.body).toEqual("Body is not a valid TechRecord");
+                    done();
+                  });
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  describe("updateTechRecords", () => {
+    context("when database is populated", () => {
+      beforeAll(async () => {
+        await emptyDatabase();
+      });
+
+      afterAll(async () => {
+        await populateDatabase();
+      });
 
+      context("when database is populated", () => {
+        const convertToResponse = (dbObj: any) => { // Needed to convert an object from the database to a response object
+          const responseObj = Object.assign({}, dbObj);
+
+          // Adding primary and secondary VRMs in the same array
+          const vrms: any = [{isPrimary: true}];
+          if (responseObj.primaryVrm) {
+            vrms[0].vrm = responseObj.primaryVrm;
+          }
+
+          Object.assign(responseObj, {
+            vrms
+          });
+
+          // Cleaning up unneeded properties
+          delete responseObj.primaryVrm; // No longer needed
```

Comment:
Not for now, but should be fixed at source. Want test data to most accurately reflect real data.

Reply:
Yeah, I don't understand why do we need to alter the original data. It looks like those properties are not being used on the client side and we just get rid of them when sending back the response. Maybe it was a fix needed by the mobile app, i can't really know